### PR TITLE
minesweeper & whiteboard

### PR DESCRIPTION
**Pull request by sam-halfpenny**

### CODE REVIEW

File: MINESWEEPER/minesweeper.js [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+let proxblocks=[]
+let mines=[]
+let uncovered=[]
+let minesweeperpic=document.getElementById('minesweeper')
+kill=false
+window.addEventListener("keydown", function(e) {
+    // space and arrow keys
+    if([32, 37, 38, 39, 40].indexOf(e.keyCode) > -1) {
+        e.preventDefault();
+    }
```
Comment (will-0) [inline MINESWEEPER/minesweeper.js:17]: What's the purpose of this?

File: MINESWEEPER/minesweeper.js [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+                case 40:
+                    minesweeper.movedown()
+                    break
+                case 32:
+                    minesweeper.dig()
+                    break
+            }
+        });
+    }
+}
```
Comment (will-0) [inline MINESWEEPER/minesweeper.js:315]: For some reason, this seems to run slowly. Can you think of any ways to speed it up?

Comment (will-0) [reply MINESWEEPER/minesweeper.js:315]: (think about why it will be going slow)

**Review by will-0 — APPROVED** [review]
Both of these are impressive projects, well done!

**Comment by will-0** [comment]
From now on, I'll leave it to you to merge the pull request. That way I can comment then approve your changes (as long as they don't break anything), and leave it up to you whether you want to make further changes before you publish it :)

File: MINESWEEPER/minesweeper.js (hunk 1 above)
Comment (sam-halfpenny) [reply MINESWEEPER/minesweeper.js:17]: this is to stop the screen from scrolling when you move

File: MINESWEEPER/minesweeper.js (hunk 2 above)
Comment (sam-halfpenny) [reply MINESWEEPER/minesweeper.js:299]: yep i figured it out i was using a useless trig function to make straight lines :b

**Pull request merged**